android/src/main/java/com/appgoalz/rnjwplayer/RNJWPlayer.java

```diff
@@ -13,6 +14,16 @@ public RNJWPlayer(Context var1, PlayerConfig var2) {
         super(var1, var2);
     }
 
+    @Override
+    public boolean dispatchKeyEvent(KeyEvent event) {
+        // Exit fullscreen or perform the action requested
+        if (event.getKeyCode() == KeyEvent.KEYCODE_BACK) {
+            this.setFullscreen(false,false);
```

Comment:
Add another if check
If(player.getFullScreen())

And only then set full screen to false. 
Also return true in this case, false otherwise. 
Because if you return false it will not consume back button and will leave the entire screen if let's say you're using react navigation